src/6_branch.js

```diff
@@ -1965,3 +1965,36 @@ Branch.prototype['trackCommerceEvent'] = wrap(callback_params.CALLBACK_ERR, func
 	});
 	done();
 });
+/*
+ logEvent() documentation
+ */
+Branch.prototype['logEvent'] = wrap(callback_params.CALLBACK_ERR, function(done, name, eventAndCustomData, contentItems) {
+	name = name && utils.validateParameterType(name, 'string') ? name : null;
```

Comment:
you should just check if the parameter exists within the `validateParameterType` function. then you can remove the additional checks